src/olympia/addons/cron.py

```diff
@@ -199,33 +199,39 @@ def deliver_hotness():
     threshold = 250 if addon type is theme, else 1000
     hotness = (a-b) / b if a > threshold and b > 1 else 0
     """
-    frozen = set(f.id for f in FrozenAddon.objects.all())
-    all_ids = list((Addon.objects.filter(status__in=amo.REVIEWED_STATUSES)
-                   .values_list('id', flat=True)))
-    now = datetime.now()
-    one_week = now - timedelta(days=7)
-    four_weeks = now - timedelta(days=28)
-
-    for ids in chunked(all_ids, 300):
-        addons = Addon.objects.filter(id__in=ids).no_transforms()
-        ids = [a.id for a in addons if a.id not in frozen]
-        qs = (UpdateCount.objects.filter(addon__in=ids)
-              .values_list('addon').annotate(Avg('count')))
-        thisweek = dict(qs.filter(date__gte=one_week))
-        threeweek = dict(qs.filter(date__range=(four_weeks, one_week)))
-        for addon in addons:
-            this, three = thisweek.get(addon.id, 0), threeweek.get(addon.id, 0)
-
-            # Update the hotness score but only update hotness if necessary.
-            # We don't want to cause unnecessary re-indexes
-            threshold = 250 if addon.type == amo.ADDON_STATICTHEME else 1000
-            if this > threshold and three > 1:
-                hotness = (this - three) / float(three)
-                if addon.hotness != hotness:
-                    addon.update(hotness=(this - three) / float(three))
-            else:
-                if addon.hotness != 0:
-                    addon.update(hotness=0)
+    frozen = set(f.addon_id for f in FrozenAddon.objects.all())
+    averages = get_averages_by_addon_from_bigquery(today=date.today())
+
+    for average in averages:
+        # BigQuery returns GUIDs
+        guid = average['addon_id']
```

Comment:
there's a risk of hitting the `guid is None` problem again here.

Reply:
The `get_averages_by_addon_from_bigquery` function omits `None` GUIDs.

Reply:
oh, I missed it because it was still called `addon_id` at that point

Reply:
I wasn't sure about renaming this property, but looks like I should.